### ISSUE TITLE
How could I do more regularization with stsb training?

### ISSUE THREAD

**Issue by PhilipMay** [UKPLab/sentence-transformers#846]
Hi @nreimers 
when I train a model on stsb I would do some more regularization.
When training on a classification task I could just increase the dropout of the prediction head.
But this models do not have a head...

On the other habd there is `attention_probs_dropout_prob` and `hidden_dropout_prob`.
But that is part of the actual language model. Changing that would destroy everything the
model already learned - I guess...

Do you know other ways to add more regularization or dropout?
Add a "naked" dropout layer on top of the last layer?

Thanks
Philip

**Comment by PhilipMay** [comment]
Hey @Aaronsom - that would be awesome! :-)

**Comment by Aaronsom** [comment]
Hi,

so quick context what my situation was: 
I was fine-tuning an image+text embedding model with triplet loss (using https://github.com/UKPLab/MMT-Retrieval which builds upon sentence-transformer) with very little training examples (only 5000 ) and I had some problems with overfitting.

I tried three different kinds of Dropout:
1) dropping entire input tokens
2) dropping entire output tokens before embedding are computed
3) normal dropout where I increase the hidden_dropout_prob from the default 0.1 probability

Of the three options, only 3) was really useful in the end, at least for me and my use-case.
So I recommend experimenting with that.

If you want to try the other two, 1) is annoying to implement and required changes to the Transformer code itself - not recommended. 2) can be done by including an additional custom module on top that does the dropping of tokens.

**Comment by PhilipMay** [comment]
Hey @Aaronsom - thanks for the quick answer.
It is so cool how fast you and also @nreimers are resonding. I love that!

PS: Did you try increasing attention_probs_dropout_prob or just hidden_dropout_prob?

**Comment by Aaronsom** [comment]
I used only hidden_dropout_prob.

**Comment by PhilipMay** [comment]
Many thanks. Since this is answered I will close it again. If I get some positive results with stsb I will let you know...
